internal/client/config.go

```diff
@@ -341,6 +370,30 @@ func (c *Config) ConfigAccess() (clientcmd.ConfigAccess, error) {
 	return c.clientConfig().ConfigAccess(), nil
 }
 
+func (c *Config) newInclusterContext() *api.Context {
+	ns, _, _ := c.clientConfig().Namespace()
+	if ns == "" {
+		ns = "default"
+	}
+	nc := api.NewContext()
```

Comment:
nit: could just build the object and return it i.e return api.Context{Cluster: inCluster, Namespace: ns, ...}

Reply:
Ok.